lib/schemes/refresh.js

```diff
@@ -0,0 +1,382 @@
+        const _tokenTTL = getProp(result, this.options.expiresIn) || this.options.token.maxAge
+        const tokenExpiration = _tokenCreatedAt + (_tokenTTL * 1000)
+        this._setTokenExpiration(tokenExpiration)
+
+        // Update refresh token expiration
+        const refreshTokenMaxAge = this.options.refreshToken.maxAge
+        if (refreshTokenMaxAge) {
+          if (!this._getRefreshTokenExpiration() || this.hasRefreshTokenChanged) {
+            this._setRefreshTokenExpiration(Date.now() + (refreshTokenMaxAge * 1000))
+            this.hasRefreshTokenChanged = false
+          }
+        }
+      }
+
+      // Update client id
+      const clientId = getProp(result, this.options.clientId)
+      if (clientId) {
+        this.$auth.setClientId(this.name, clientId)
+      }
+
+      // End the refresh
+      this.isRefreshing = false
+    }
+  }
+
+  async _tokenRefresh () {
+    // Refresh endpoint is disabled
+    if (!this.options.endpoints.refresh) return
+
+    // Sync tokens
+    this._setToken(this.$auth.syncToken(this.name))
+    this._syncRefreshToken()
+
+    // Token is required but not available
+    if (this.options.tokenRequired && !this.$auth.getToken(this.name)) return
+
+    // Token is already being refreshed
+    if (this.isRefreshing) return
+
+    // Start the refresh
+    this.isRefreshing = true
+
+    let { dataClientId, dataGrantType, grantType } = this.options
+    const endpoint = {
+      data: {
+        [this.options.dataRefreshToken]: this._getRefreshToken()
+      }
+    }
+
+    // Only add client id to payload if enabled
+    if (dataClientId) {
+      endpoint.data[dataClientId] = this.$auth.getClientId(this.name)
+    }
+
+    // Only add grant type to payload if enabled
+    if (dataGrantType !== false) {
+      endpoint.data[dataGrantType] = grantType
+    }
+
+    // Try to fetch user and then set
+    return this.$auth.requestWith(
+      this.name,
+      endpoint,
+      this.options.endpoints.refresh
+    ).then(response => {
+      this._updateTokens(response)
+    }).catch(() => {
+      // TODO: Unhandled error
+      this._logoutLocally()
+    })
+  }
+
+  _scheduleTokenRefresh () {
+    // If auto refresh is disabled, bail
+    if (!this.options.autoRefresh.enable) return
+
+    let intervalDuration = (this._getTokenExpiration() - Date.now()) * 0.75
+    if (intervalDuration < 1000) {
+      // in case you misconfigured refreshing this will save your auth-server from a self-induced DDoS-Attack
+      intervalDuration = 1000
+    }
+
+    this.refreshInterval = setInterval(() => {
+      this._tokenRefresh()
+    }, intervalDuration)
+  }
+
+  _initializeRequestInterceptor () {
+    this.$auth.ctx.app.$axios.onRequest(async config => {
+      // Sync tokens
+      const token = this.$auth.syncToken(this.name) || false
+      const refreshToken = this._syncRefreshToken()
+      this._setToken(token)
+
+      // Update header
+      config.headers[this.options.tokenName] = token
```

Comment:
You should move this line after the if of token check as we get a false value from the header request if no token present.